internal/pkg/aws/elbv2/elbv2.go

```diff
@@ -33,6 +35,44 @@ func New(sess *session.Session) *ELBV2 {
 	}
 }
 
+// ListenerRuleHostHeaders returns all the host headers for a listener rule.
+func (e *ELBV2) ListenerRuleHostHeaders(ruleARN string) ([]string, error) {
+	resp, err := e.client.DescribeRules(&elbv2.DescribeRulesInput{
+		RuleArns: aws.StringSlice([]string{ruleARN}),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("get listener rule for %s: %w", ruleARN, err)
+	}
+	if len(resp.Rules) == 0 {
+		// Ideally this should never happen.
+		return nil, fmt.Errorf("cannot find listener rule %s", ruleARN)
+	}
+	rule := resp.Rules[0]
+	hostHeaderSet := make(map[string]bool)
+	for _, condition := range rule.Conditions {
+		if aws.StringValue(condition.Field) == "host-header" {
+			for _, value := range condition.Values {
+				hostHeaderSet[aws.StringValue(value)] = true
+			}
```

Comment:
Wut, what's the difference between `condition.Values` and `condition.HostHeaderConfig.Values`?
If I look at our CFN templates, we only have the latter.

Reply:
Yeah but the results always have both two. I check values just to be safe...

Reply:
OK I just looked this up lol and these field names are so confusing 🤦 

`condition.Values` allows you to specify only a single host name although the type is an array.
`condition.HostHeaderConfig` allows you to specify multiple host names.

So yeah, we should check both 👍 , maybe drop a comment with an explanation?
```suggestion
			// Values is a legacy field that allowed specifying only a single host name.
			// The alternative is to use HostHeaderConfig for multiple values.
			// Only one of these fields should be set, but we collect from both to be safe.
			for _, value := range condition.Values {
				hostHeaderSet[aws.StringValue(value)] = true
			}
```